tests/test_spatial.py

```diff
@@ -124,128 +210,171 @@ def test_chunked_spatial_average_for_lat_region(self):
+            dims=["lat", "lon"],
+        )
 
-        with pytest.raises(TypeError):
-            self.ds.spatial._validate_region_bounds("lon", "str")
+        xr.testing.assert_allclose(result, expected)
 
-    def test_raises_error_if_there_are_0_elements_in_the_bounds(self):
-        with pytest.raises(ValueError):
-            self.ds.spatial._validate_region_bounds("lon", ())
+    def test_weights_for_region_in_lon_domain(self):
+        expected = xr.DataArray(
+            data=np.array(
+                [
+                    [0.0, 0.0, 0.00297475, 0.0],
+                    [0.0, 0.0, 49.99702525, 0.0],
+                    [0.0, 0.0, 49.99702525, 0.0],
+                    [0.0, 0.0, 0.00297475, 0.0],
+                ]
+            ),
+            coords={"lat": self.ds.lat, "lon": self.ds.lon},
+            dims=["lat", "lon"],
+        )
+        result = self.ds.spatial.get_weights(
+            axis=["Y", "X"], lat_bounds=None, lon_bounds=(-170, -120)
+        )
 
-    def test_raises_error_if_there_are_more_than_two_elements_in_the_bounds(self):
-        with pytest.raises(ValueError):
-            self.ds.spatial._validate_region_bounds("lon", (1, 1, 2))
+        xr.testing.assert_allclose(result, expected)
 
-    def test_does_not_raise_error_if_lower_and_upper_bounds_are_floats_or_ints(self):
-        self.ds.spatial._validate_region_bounds("lon", (1, 1))
-        self.ds.spatial._validate_region_bounds("lon", (1, 1.2))
+    def test_weights_for_region_in_lon_domain_with_both_spanning_p_meridian(self):
+        # FIXME: ValueError when domain or region bounds contains lower bound
+        # larger than upper bound
+        ds = self.ds.copy()
+        # Domain spans prime meridian.
+        ds.lon_bnds.data[:] = np.array([[359, 1], [1, 90], [90, 180], [180, 359]])
 
-    def test_raises_error_if_lower_bound_is_not_a_float_or_int(self):
-        with pytest.raises(TypeError):
-            self.ds.spatial._validate_region_bounds("Y", ("invalid", 1))
+        result = ds.spatial.get_weights(
+            axis=["X"],
+            # Region spans prime meridian.
+            lat_bounds=None,
+            lon_bounds=np.array([359, 1]),
+        )
+        expected = xr.DataArray(
+            data=np.array([2.0, 0.0, 0.0, 0.0]),
+            coords={"lon": ds.lon},
+            dims=["lon"],
+        )
 
-    def test_raises_error_if_upper_bound_is_not_a_float_or_int(self):
-        with pytest.raises(TypeError):
-            self.ds.spatial._validate_region_bounds("X", (1, "invalid"))
+        xr.testing.assert_allclose(result, expected)
 
-    def test_raises_error_if_lower_lat_bound_is_bigger_than_upper(self):
-        with pytest.raises(ValueError):
-            self.ds.spatial._validate_region_bounds("Y", (2, 1))
+    def test_weights_for_region_in_lon_domain_with_domain_spanning_p_meridian(self):
+        # FIXME: ValueError when domain bounds contains lower bound larger than
+        # upper bound
```

Comment:
I think this should throw an error? I'm kind of confused why it wasn't before based on [this check](https://github.com/xCDAT/xcdat/blob/573c4ad361107f644184d2b9ae614d5b26d90d65/xcdat/spatial.py#L375). Should this test have region bounds such that `lon_bounds=np.array([350, 20])` or something like that? 

I think region bounds should be able to accept a larger right hand bound, but I don't think this should be true for domain bounds.

Reply:
Okay I understand now, domain bounds should not have a larger right hand value so it should throw an error.

The old test below was faulty in that it didn't throw an error for this case, since we skipped the `_validate_domain_bounds()` check by directly testing this case with `_get_longitude_bounds()` instead of `get_weights()`
https://github.com/xCDAT/xcdat/blob/223e73b60f19b2634a57fd0a5ce5105fda3fb1f6/tests/test_spatial.py#L494-L510

Basically, this test should be deleted.

Reply:
I ended up deleting this test because it is faulty and we already have a test for raising a ValueError if the domain bounds has a larger lower bound vs. upper bound.